_posts/2021-07-13-gs_00003_terminal_basics2.md

```diff
@@ -0,0 +1,394 @@
+another example and have a closer look at the details.
+
+You can open a firefox browser with:
+
+    $ firefox
+
+<div class='note-normal' markdown="1">
+_Note: You also can install a browser of your choice using Pacman.
+Firefox is preinstalled with Manjaro, so we use it here._
+</div>
+
+A firefox window should open. If you want to stop firefox, you can either close
+the window or press CTRL-C inside the terminal. It will also stop if you close
+the terminal. Just have a try!
+
+In the case you stopped firefox, start it again with the command above and take
+a look at the terminal. The cursor moved to the next line, but the line is
+blank. If you try to type a command like "ls" you will note that you can type
+it, but it does not do anything. This is because you can only execute a new
+command after the previous one is completed. Here, firefox is still running. An
+easy workaround is to open a second terminal and run the "ls" command there.
+Have a try! It will work. However, there is something else you can do. Stop
+firefox with CTRL-C and run it, again, with a tiny change:
+
+    $ firefox &
+
+The "&" indicates that firefox shall run in the background. You will be prompted
+a number, which is the so-called **process id**, and you are able to enter
+further commands. If you want to stop firefox from the command line, you can
+kill it using one of the following:
+
+    $ kill <process id>
+    $ killall firefox
+
+The process id is the one you were shown when starting firefox. The "kill"
+command will, therefore, only kill that specific firefox process. The "killall"
+command will kill all firefox instances. You might wonder if you really need to
+take note of the process id, and the answer is, of course, no! Always remember
+that programmers are lazy! You can use the "ps" command to find the process id
+(PID).
+
+    $ ps
+
+There are many options you can add to the command. Check the man page for a full
+list. An often used command is
+
+    $ ps aux
+
+which gives a list of all running processes. Since that list is usually long,
+you can search for a specific process:
+
+    $ ps aux | grep firefox
+
+You should be able to find your PID and also might see that firefox created
+process children. If you kill the main process, you will kill all children as
+well. Damn ... This world is so cruel! Sometimes it is also nice to use:
+
+    $ ps axjf
+
+This will show you a process tree. You can see that firefox is a child of
+something called "bash". We will talk about it in the next post, but it means
+your terminal. Again, you can see the children of firefox.
+
+One last thing! Sometimes, it is useful to monitor the CPU utilization and see
+which process uses it. A nice tool to do so is:
+
+    $ htop
+
+Our computer uses right now 814 MB of RAM and has a CPU utilization of around
+20%. What about yours? If you have multiple cores, you can see the utilization
+per core. Exit "htop" using CTRL-C.
```

Comment:
Maybe add something like:
"By the way, CTRL-C is commonly used in the terminal to exit running programs."

Reply:
to be changed

Reply:
I added it in a previous section as note